keras/src/layers/attention/multi_head_attention.py

```diff
@@ -392,7 +399,13 @@ def _masked_softmax(self, attention_scores, attention_mask=None):
         return self._softmax(attention_scores, mask=attention_mask)
 
     def _compute_attention(
-        self, query, key, value, attention_mask=None, training=None
+        self,
+        query,
+        key,
+        value,
+        return_attention_scores,
```

Comment:
@divyashreepathihalli shouldn't `return_attention_scores` have a default value here?
There might be many instances of `_compute_attention` call that don't pass `return_attention_scores` value and I think this change can break them. There is an example [here](https://github.com/keras-team/keras-hub/blob/5a7ecb644fe3ec5b04519846c2c5c54ad58fbe12/keras_hub/src/layers/modeling/cached_multi_head_attention.py#L111-L117). So I was wondering if it's possible to set a default value here so that other references of `_compute_attention` work as before.
I think this change is probably the reason that the test is failing in https://github.com/keras-team/keras-hub/pull/1977

Reply:
Isn't there any other way to check and make sure the users pass the arg value (rather than lack of default value here)?

Reply:
I think a work around would be to add a `self._return_attention_scores` and then set it in the call method and use it in `_compute_attention`. wdyt?